greenplumpython/db.py

```diff
@@ -79,8 +79,21 @@ def table(self, name: str):
 
         return table(name, self)
 
-    def apply(self, func: Callable[[], "FunctionExpr"]) -> "FunctionExpr":
-        return func().bind(db=self)
+    def assign(self, **new_columns: Callable[[], Any]) -> "Table":
+        from greenplumpython.expr import Expr
+        from greenplumpython.func import FunctionExpr
+        from greenplumpython.table import Table
+        from greenplumpython.type import to_pg_const
+
+        targets: List[str] = []
+        for k, f in new_columns.items():
+            v: Any = f()
+            if isinstance(v, Expr):
+                assert v.table is None, "New column should not depend on any table."
+            if isinstance(v, FunctionExpr):
+                v = v.bind(db=self)
+            targets.append(f"{v.serialize() if isinstance(v, Expr) else to_pg_const(v)} AS {k}")
```

Comment:
So we are supposing the result column name is compulsory when calling `assign()` like `assign(result_col=lambda)`